spec/bolt_spec/run_spec.rb

```diff
@@ -69,16 +69,70 @@
 
   describe 'run_plan' do
     it 'should run a plan' do
-      result = run_plan('sample::single_task', { 'nodes' => 'ssh' }, config: config_data, inventory: inventory_data)
+      result = run_plan('sample::single_task', { 'nodes' => 'ssh' }, options)
       expect(result['status']).to eq('success')
       data = result['value'][0]
       expect(data['status']).to eq('success')
     end
 
     it 'should return a failure' do
-      result = run_plan('error::run_fail', { 'target' => 'ssh' }, config: config_data, inventory: inventory_data)
+      result = run_plan('error::run_fail', { 'target' => 'ssh' }, options)
       expect(result['status']).to eq('failure')
       expect(result['value']['kind']).to eq('bolt/run-failure')
     end
   end
+
+  context 'with a target that has a puppet-agent installed' do
+    def run_root_config
+      { 'ssh' => root_config['ssh'] }
+    end
+
+    before(:all) do
+      result = run_task('puppet_agent::version', 'ssh', inventory: conn_inventory, config: run_root_config)
+      expect(result.first['status']).to eq('success')
+      unless result.first['result']['version']
```

Comment:
I encountered a bug in the `puppet_agent::install`/`facts` tasks when running against the `rastasheep/ubuntu-sshd:16.04` container we build off of for `bolt-spec`.

When facter is not installed the `facts::bash` task determines the platform and release for 16.04 as expected with 
```
root@2b504359b3b0:/# grep "^NAME" /etc/os-release | cut -d'=' -f2 | sed "s/\"//g"
Ubuntu
root@2b504359b3b0:/# grep "^VERSION_ID" /etc/os-release | cut -d'=' -f2 | sed "s/\"//g"
16.04
```
The results of running facter are different than those using the shell code.
```
root@2b504359b3b0:/# /opt/puppetlabs/bin/facter --yaml os.name
os.name: Debian
root@2b504359b3b0:/# /opt/puppetlabs/bin/facter --yaml os.release.full
os.release.full: stretch/sid
```
This causes an error in the constructing the target download of the `.deb`. 
```
expected: http://apt.puppetlabs.com/puppet-release-xenial.deb
observed: http://apt.puppetlabs.com/puppet-release-.deb
```
Looks like this has been reported as well at https://tickets.puppetlabs.com/browse/BOLT-997

Reply:
Returning Debian on Ubuntu sounds very wrong.